### ISSUE TITLE
Oddities with default inventory/wield images for nodes

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#9209]
##### Minetest version
```
5.1.0
```

##### OS / Hardware
Operating system: GNU/Linux

##### Summary
This is partially bug report, partially feature request.

There are some oddities, inconsistencies and outright bugs with the default inventory/wield images for nodes in their item form. That is, how the item looks like (in inventory/when wielded/when dropped as an item entity) when `inventory_image` or `wield_image` are not set in the item definition.

##### Actual behavior

Bad inventory images:

* `airlike`, `torchlike`, `signlike`: Invisible
* `plantlike`: Plant appears as extruded 2D sprite
* `plantlike_rooted`: Same as `plantlike`. Also, the “base cube” of the node does not appear in the inventory image at all

Bad wield images:

* `airlike`: Wielded item goes completely bonkers. It constantly changes in size, shape and even the texture.
* `raillike`, `torchlike`: Practically invisible. But if you move the camera like crazy and jump, you might see it for a split second, in an odd angle. So it seems the wielded item is there, it's just off-camera
* `signlike`: Item is in a weird angle
* `plantlike`, `plantlike_rooted`: Same problem as for their inventory images

Especially invisible wield/inventory items can be quite nasty as you can't see them in your inventory. When those items drop, they can be really nasty to track down.

##### Steps to reproduce
Use a test mode with some dummy nodes of the drawtypes given above and `inventory_image` and `wield_image` set to `nil`. You can use DevToys as template:
https://forum.minetest.net/viewtopic.php?f=9&t=23802

(But you still need to edit the node definitions)

##### Suggested fixes:

First of all, here's what I expect: That the default appearance of items in their inventory looks like their 3D form. This is because leaving `inventory_image` and `wield_image` at `nil` is currently the [i]only[/i] way to get 3D images at all. Setting `inventory_image` or `wield_image` explicitly will revert to 2D. So, it's reasonable to offer modders the *option* of using the default for the 3D form. From this standard a deviation is only allowed if the default appearance would somehow look broken.

That having said, here's the concrete suggestions:

* `airlike`: Add dummy inventory and wield image (similar to what we have for the `air` node; but should use different texture)
* `raillike`: Rotate wield image properly.
* `signlike`: Rotate wield image properly. Fix invisible inventory image
* `torchlike`: Use “standing” torch texture and make item visible in inventory/wieldhand
* `plantlike`: Make inventory image look like the plant mesh. Use the “X” shape by default. If `paramtype2=="meshoptions"` and `place_param2` are set, change the inventory/wield mesh to the corresponding style
* `plantlike_rooted`: Similar to `plantlike` for the “plant” part of the node, but also render the “base” cube. For the inventory image, center camera on plant

**Comment by paramat** [comment]
> airlike, torchlike, signlike: Invisible

airlike is invisible, but i doubt the others are, do you mean no backface when rotating in inventory?
> plantlike: Plant appears as extruded 2D sprite

I think this is acceptable.
> plantlike_rooted: [...] Also, the “base cube” of the node does not appear in the inventory image at all

Not a big issue. Some uses of rooted consider it to be the top section only, kelp does this.

Feedback on your suggestions: All fine except:
> plantlike: [...] If paramtype2=="meshoptions" and place_param2 are set, change the inventory/wield mesh to the corresponding style

-1, too complex and rather unnecessary, also visually less clear than an extruded single tile.
> plantlike_rooted: [...] also render the “base” cube

Not a big issue, see above.

**Comment by Wuzzy2** [comment]
The issue with `airlike` is that it's *item form* is invisible, not that it's node form is invisible (that's the point). They are invisible in inventory and in dropped item form, which is pretty annoying to debug should such items somehow sneak into the world (due to a bug in the mod code). The PR will add a simple fallback image.

Missing backfaces is a problem for sprite-based nodes such as `firelike` or `torchlike` as sprites are meant to have textures on both sides. It's the reason why some some item images appeared invisible. It also made `firelike` nodes appear weird in inventory. It's also a problem for the dropped item form or with rotating items. All of this is fixed in PR.

I have independently come to the conclusion the `plantlike` nodes should be left as-is. I have tested my proposal but it looked worse than before. But well, it was an interesting experiment.

I have not done `place_param2` in the PR simply because I did not manage to access it in C++. :D
And yes, it's a little bit overkill.

About `plantlike_rooted`: Yeah, the idea of this issue is to make most item images represent the node as closely as possible *in their default form*. I don't really have any good idea for `plantlike_rooted`, it's somewhat special.
